creator/velociprobe.py

```diff
@@ -0,0 +1,136 @@
+import h5py
+import numpy as np
+
+from .nx_creator_ptycho import NXCreator
+
+
+def velociprobe2nexus(master_path, position_path, nexus_path):
+    """Convert APS velociprobe data to the new Nexus format.
+
+    Because the Velociprobe is collected with a Dectris Eiger detector the
+    master file is very Nexus-like. It claims to follow the NXmx standard for
+    macromolecular crystallography, but it also contains a bunch of nonsense
+    in the NXTransformations.
+    """
+
+    with h5py.File(master_path, 'r') as f, NXCreator(nexus_path) as creator:
+
+        entry = creator.create_entry_group(definition='basic')
+
+        instrument = creator.create_instrument_group(entry=entry)
```

Comment:
Naming these parameters all 'h5parent' seems less 'self-documenting' when compared with naming them after the specific parent group they should be placed in.

Reply:
Ah, I see you chose "entry" as argument name, since most of them will have the entry group as h5parent. What about groups that don't go under entry, e.g. detector - should that be called "instrument" instead? And what about the transformation group that either has sample or detector groups as parent?

Reply:
In NeXus, the NXinstrument group is a child of the NXentry group, so this default is the most common.  Alternatives would include `entry="entry1"`, etc.)

Reply:
Sure, what do you suggest now for the naming of the argument variable (not the specific input)?

Reply:
I'm ok with the `entry` keyword as a name for the `create_instrument_group()` method.

Reply:
Sure, and then for create_detector_group we use a different one? I mean it does not really matter it's just a name. I only chose because that is what it is and it's general enough to be the same for all creator methods.

Reply:
I was proposing that we use "h5parent" when the hierarchy is unknown such as for transformation, positioner groups, but when the hierarchy is known we use the name of the parent ("entry", "instrument") for groups such as instrument, beam, detector, sample, axis.

Similarly (but independently) we can also enforce the correct hierarchy by checking that the NX_class attribute of the parent is acceptable.

Reply:
yeah, I see pro and con for both approaches- naming all parent args _h5parent_ or only the ones where the parent is ambiguous and fixing others like instrument, beam etc. I'd prefer to name them all _h5parent_ and describe the default in doc, but happy to follow majority decision.

Yes let's check the NX_class attributes in addition no matter which convention we follow.